content/posts/networkx/node-ordering-Ti-updating/index.md

```diff
@@ -0,0 +1,91 @@
+---
+title: "Updates on VF2++"
+date: 2022-07-06
+draft: false
+description: "
+Summary of the progress on VF2++
+"
+tags: ["gsoc", "networkx"]
+displayInList: true
+author: ["Konstantinos Petridis"]
+
+---
+
+This post includes all the major updates since the last post about VF2++. Each section is dedicated to a different
+sub-problem and presents the progress on it so far. General progress, milestones and related issues can be
+[found here](https://github.com/kpetridis24/networkx/milestone/1).
+
+## Node ordering
+
+The node ordering is one major modification that **VF2++** proposes. Basically the nodes are examined in an order that
+makes the matching faster, by firstly examining nodes that are more likely to match. This part of the algorithm has been
+implemented, however there is an issue. The existence of detached nodes (not connected to the rest of the graph) causes
+the code to crash. Fixing this bug will be a top priority during the next steps. The ordering implementation is described
+by the following pseudocode.
+
+> ---
+>
+> **Matching Order**
+>
+> ---
+>
+> 1. **Set** $M = \varnothing$.
+> 2. **Set** $\bar{V1}$ : nodes not in order yet
+> 3. **while** $\bar{V1}$ not empty **do**
+>    - $rareNodes=\[$nodes from $V_1$ with the rarest labels$\]$
+>    - $maxNode=argmax_{degree}\(rareNodes\)$
+>    - $T=$ BFSTree with $maxNode$ as root
+>    - **for** every level in $T$ **do**
+>      - $V_d=\[$nodes of the $d^{th}$ level$\]$
+>      - $\bar{V_1} \setminus V_d$
+>      - $ProcessLevel(V_d)$
+> 4. Output $M$: the matching order of the nodes.
+
+> ---
+>
+> **Process Level**
+>
+> ---
+>
+> 1. **while** $V_d$ not empty **do**
+>    - $S=\[$nodes from $V_d$ with the most neighbors in M$\]$
+>    - $maxNodes=argmax_{degree}\(S\)$
+>    - $rarestNode=\[$node from $maxNodes$ with the rarest label$\]$
+>    - $V_d \setminus m$
+>    - Append m to M
+
+## $T_i$ and $\tilde{T_i}$
+
+According to the VF2++ paper notation:
+
+$$T_1=\(u\in V_1 \setminus m: \exists \tilde{u} \in m: (u,\tilde{u}\in E_1)\)$$
+
+where $V_1$ and $E_1$ contain all the nodes and edges of the first graph respectively, and $m$ is a dictionary, mapping
+every node of the first graph to a node of the second graph. Now if we interpet the above equation, we conclude that
+$T_1$ contains uncovered neighbors of covered nodes. In simple terms, it includes all the nodes that do not belong in
+the mapping $m$ yet, but are neighbors of nodes that are in the mapping. In addition,
+
+$$\tilde{T_1}=\(V_1 \setminus m \setminus T_1\)$$
+
+The following figure is meant to provide some visual explanation of what exactly $T_i$ is.
+
+<center><img src="Ti.png" alt="Illustration of $T_i$"/></center>
```

Comment:
Is there a reason you did not use normal markdown here?
```suggestion
![Illustration of $T_i$.](Ti.png)
```

Reply:
nope, i just took it as is from another post. I am not an md expert unfortunately :)